src/main/java/core/basesyntax/model/zoo/Animal.java

```diff
@@ -1,9 +1,31 @@
 package core.basesyntax.model.zoo;
 
+import javax.persistence.DiscriminatorColumn;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+
+@Entity
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@DiscriminatorColumn(name = "animal")
```

Comment:
Probably it is better to specify the `DiscriminatorType` in order to avoid ambiguous types